Sources/Valkey/Subscriptions/ValkeyConnection+subscribe.swift

```diff
@@ -185,18 +167,40 @@ extension ValkeyConnection {
     /// channel
     ///
     /// - Parameters:
+    ///   - isolation: Actor isolation
     ///   - process: Closure that is called with async sequence of key invalidations
     /// - Returns: Return value of closure
     @inlinable
     public func subscribeKeyInvalidations<Value>(
+        isolation: isolated (any Actor)? = #isolation,
         process: (AsyncMapSequence<ValkeySubscription, ValkeyKey>) async throws -> sending Value
     ) async throws -> sending Value {
-        try await self.subscribe(to: [ValkeySubscriptions.invalidateChannel]) { subscription in
+        try await self.subscribe(to: [ValkeySubscriptions.invalidateChannel], isolation: isolation) { subscription in
             let keys = subscription.map { ValkeyKey($0.message) }
             return try await process(keys)
         }
     }
 
+    @inlinable
+    func subscribe<Value>(
+        command: some ValkeyCommand,
+        filters: [ValkeySubscriptionFilter],
+        isolation: isolated (any Actor)? = #isolation,
+        process: (ValkeySubscription) async throws -> sending Value
+    ) async throws -> sending Value {
+        let (id, stream) = try await subscribe(command: command, filters: filters)
+        let value: Value
+        do {
+            value = try await process(stream)
+            try Task.checkCancellation()
```

Comment:
I think it becomes borderline impossible to exit this closure without CancellationError, is this the  behavior we want?

Reply:
Moved the unsubscribe into its own unstructured Task